File: android/src/main/kotlin/dev/gece/imaplayer/ImaPlayerPlugin.kt
package dev.gece.imaplayer

import androidx.annotation.NonNull
import io.flutter.embedding.engine.plugins.FlutterPlugin

/** ImaPlugin */
class ImaPlayerPlugin : FlutterPlugin {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {

        binding.platformViewRegistry.registerViewFactory(
            "gece.dev/imaplayer", ImaPlayerViewFactory(binding.binaryMessenger)
        )
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {

    }

}

File: ios/Classes/ImaPlayerPlugin.swift
import Flutter
import UIKit

public class ImaPlayerPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        registrar.register(
            ImaPlayerViewFactory(registrar: registrar),
            withId: "gece.dev/imaplayer_view"
        )
    }
}
